home/tests.py

```diff
@@ -96,3 +95,97 @@ def question_tag_test_data(question_test_data):
+
+        @pytest.mark.django_db
+        def test_thumb_down_answer(self, answers):
+            """
+            thumb_down_answer(answer_id) increase the thumbs down field of an answer by one
+            """
+            prev_thumb_val = answers[0].dislikes_count
+            answers[0].thumb_down_answer()
+            assert(prev_thumb_val == answers[0].dislikes_count - 1)
+
+        @pytest.mark.django_db
+        def test_set_is_edited(self, answers):
+            """
+            Tests functionality of set_is_edited
+            """
+            prev_is_edited_val = answers[0].is_edited
+            answers[0].set_is_edited(not prev_is_edited_val)
+            assert(prev_is_edited_val != answers[0].is_edited)
+
+        @pytest.mark.django_db
+        @pytest.mark.parametrize('filterType, expected', [('date', 'Answer 2'), ('votes', 'Answer 1')])
+        def test_answers_feed(self, filterType, expected, answers):
+            """
+            Tests if get_answers_feed returns queryset of all the answers of
+            question number 3 sorted by votes or answers
+            """
+            question = Question.objects.get(id=3)
+            answers_feed = question.get_answers_feed(filterType)
+            assert isinstance(answers_feed, QuerySet)
+            assert answers_feed[0].content == expected
+
+    class TestQuestionRelatedMethods:
+        @pytest.mark.django_db
+        def test_get_question_title(self):
+            """
+            get_question_title(question_id) returns a string for question
+            should return question subject and question title
+            """
+            question = Question.objects.get(id=1)
+            assert(question.get_question_title() == "Math-question from math course")
+
+    class TestHTMLRelated:
+        @pytest.mark.django_db
+        @pytest.fixture
+        def response(self, client):
+            url = reverse('question-detail', args=[2])
+            response = client.get(url)
+            return response
+
+        @pytest.mark.django_db
+        def test_display_question_page_url(self, response):
+            """
+            This test checks the returned status for routing to display-question feature
+            """
+            assert response.status_code == 200
+
+        @pytest.mark.django_db
+        def test_template_name(self, response):
+            assert response.templates[0].name == "home/question_detail.html"
+
+        @pytest.mark.django_db
+        def test_response_context(self, response):
+            '''
+            Testing if the context passed to the view contains the right contents
+            '''
+            question = Question.objects.get(id=2)
+            context = question.make_context('')
+            checkList = ['question', 'answers', 'answersCount', 'tags', 'title']
+            for check in checkList:
+                assert str(response.context[check]) == str(context[check])
```

Comment:
this is checking the code with itself, its a useless test

if you want to ensure that function A calls function B and returns a value that is based on the value returned from function B, you need to use a mockup of function B and a sentinel value.

Mockups and sentinel values an be created with the unitest.mock library, and be injected into the application via the `monkeypatch` fixture.

But since you have very tight cntrol over the test data in the system, you can avoid doing any of that and just compare the data in the context you got to a hard-wired set of expected values.

Reply:
Fixed, I compare it now with a hard-wired set of expected values.
Please note that I use string.startswith function because the question date is created to Now() time.
I know it's problematic, but I will change it after we will have more test data.
I opened an issue: #84 for it.